bindings/pydeck/setup.py

```diff
@@ -1,232 +1,117 @@
-from __future__ import print_function
-
-from distutils.command.install import install
-from setuptools import setup, find_packages, Command
-from setuptools.command.sdist import sdist
-from setuptools.command.build_py import build_py
-from setuptools.command.egg_info import egg_info
-from setuptools.command.develop import develop
-
-import atexit
-from distutils import log
-import os
-from shutil import copy
-from subprocess import check_call, check_output
+"""
+pydeck setup
+"""
+import json
 import sys
+from pathlib import Path
 
-here = os.path.dirname(os.path.abspath(__file__))
+import setuptools
 
-version_ns = {}
-with open(os.path.join(here, "pydeck", "_version.py")) as f:
-    exec(f.read(), {}, version_ns)
+HERE = Path(__file__).parent.resolve()
+
+# Get the package info from package.json
+pkg_json = json.loads((HERE / "package.json").read_bytes())
 
+# The name of the project
+name = "pydeck"
 
-def read(*parts):
-    return open(os.path.join(here, *parts), "r").read()
+# lab_path = (HERE / pkg_json["jupyterlab"]["outputDir"])
+lab_path = HERE / "pydeck" / "labextension"
 
+# Representative files that should exist after a successful build
+ensured_targets = [str(lab_path / "package.json"), str(lab_path / "static/style.js")]
 
-log.info("setup.py entered")
-log.info("$PATH=%s" % os.environ["PATH"])
+labext_name = pkg_json["name"]
 
-PATH_TO_WIDGET = "../../modules/jupyter-widget"
-PATH_TO_REPO_ROOT = "../.."
+data_files_spec = [
+    (
+        "share/jupyter/labextensions/%s" % labext_name,
+        str(lab_path.relative_to(HERE)),
+        "**",
+    ),
+    ("share/jupyter/labextensions/%s" % labext_name, str("."), "install.json"),
+]
+
+long_description = (HERE / "README.md").read_text()
+
+version_ns = {}
+with open(HERE / "pydeck" / "_version.py") as f:
+    exec(f.read(), {}, version_ns)
 
-yarn_root = os.path.join(here, PATH_TO_REPO_ROOT)
-widget_dir = os.path.join(here, PATH_TO_WIDGET)
+version = (
+    version_ns["__version__"]
+    .replace("-alpha.", "a")
+    .replace("-beta.", "b")
+    .replace("-rc.", "rc")
+)
 
-npm_path = os.pathsep.join(
-    [os.path.join(PATH_TO_REPO_ROOT, "node_modules", ".bin"), os.environ.get("PATH", os.defpath)]
+setup_args = dict(
+    name=name,
+    version=version,
+    url=pkg_json["homepage"],
+    author=pkg_json["author"]["name"],
+    author_email=pkg_json["author"]["email"],
+    description=pkg_json["description"],
+    license=pkg_json["license"],
+    license_file="LICENSE.txt",
```

Comment:
Given that in this PR you're now defining more of the build process in `pyproject.toml`, move the non-dynamic parts of the setuptools config there too? https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html